src/structures/BaseEmoji.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const Emoji = require('./Emoji');
+
+/**
+ * Parent class for `GuildEmoji` and `GuildPreviewEmoji`.
+ * @extends {Emoji}
+ */
+class BaseEmoji extends Emoji {
+  /**
+   * @param {Client} client The instantiating client
+   * @param {Object} data The data for the emoji
+   * @param {Guild} guild The guild the emoji is a part of
+   */
+  constructor(client, data, guild) {
+    super(client, data);
+
+    /**
+     * The guild this emoji is a part of
+     * @type {Guild}
+     */
+    this.guild = guild;
+
+    /**
+     * The ID of this emoji
+     * @type {Snowflake}
+     * @name BaseEmoji#id
+     */
+
+    this._patch(data);
+  }
+
+  _clone() {
```

Comment:
Isn't this super redundant?

Reply:
Yes

Reply:
Sorry, but I want to make sure I understood correctly; the redundancy here is having that `_clone` method and calling `_patch()` in the constructor, correct?

Reply:
No, just the `_clone` function is redundant.

Reply:
Alright, will remove it now, thank you for notifying me and answering so quickly. :+1: